components/molecule/phoneValidation/src/index.js

```diff
@@ -0,0 +1,145 @@
+import {useEffect, useRef, useState} from 'react'
+
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+
+import AtomIcon, {
+  ATOM_ICON_COLORS,
+  ATOM_ICON_SIZES
+} from '@s-ui/react-atom-icon'
+import MoleculeDropdownOption from '@s-ui/react-molecule-dropdown-option'
+
+import {getFlagEmoji, phoneValidationType} from './settings.js'
+
+const BASE_CLASS = 'sui-MoleculePhoneValidation'
+
+export default function MoleculePhoneValidation({
+  dropdownIcon,
+  maxLength = 11,
+  phone,
+  prefixes = [],
+  setPhone,
+  type = phoneValidationType.DEFAULT
+}) {
+  const [showDropdown, setShowDropdown] = useState(false)
+  const [selectedPrefix, setSelectedPrefix] = useState(prefixes[0])
+  const modalRef = useRef(null)
+
+  const baseClass = cx(
+    {
+      [`splitted`]: type === phoneValidationType.SPLITTED
+    },
+    BASE_CLASS
+  )
+
+  useEffect(() => {
+    const handleClickOutside = event => {
+      if (modalRef.current && !modalRef.current.contains(event.target)) {
+        setShowDropdown(false)
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [baseClass])
+
+  useEffect(() => {
+    phone && setPhone(`${selectedPrefix.countryCode} ${phone.split(' ')[1]}`)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPrefix])
+
+  return (
+    <div className={baseClass}>
+      <div className={`${baseClass}-input`}>
+        <div
+          className={`${baseClass}-input-prefix`}
+          testId="molecule-phone-validation-input-prefix"
+          onClick={() => setShowDropdown(!showDropdown)}
+        >
+          <span className={`${baseClass}-input-prefix-flag`}>
+            {getFlagEmoji(selectedPrefix.value)}
+          </span>
+          <AtomIcon
+            size={ATOM_ICON_SIZES.medium}
+            color={ATOM_ICON_COLORS.currentColor}
+          >
+            {dropdownIcon}
+          </AtomIcon>
```

Comment:
Maybe you have investigated it but let me remind you, instead of building this dropdown list 100% from scratch, we do not have a dropdown component at SUI that covers this? If it's yes, it will be more testable than do it from scratch. 

When I click into the flag the list is opened well, but if I click again into the flag, as a user I expect that the flag list will be closed again. You have achieve it at focus out flow, but not when user clicks again to the flag.

Reply:
Omw <3